jerry-core/parser/js/js-lexer.c

```diff
@@ -2882,12 +2885,39 @@ lexer_construct_regexp_object (parser_context_t *context_p, /**< context */
   re_compiled_code_t *re_bytecode_p = re_compile_bytecode (pattern_str_p, current_flags);
   ecma_deref_ecma_string (pattern_str_p);
 
+  uint8_t literal_type = LEXER_REGEXP_LITERAL;
+
   if (JERRY_UNLIKELY (re_bytecode_p == NULL))
   {
+#if ENABLED (JERRY_ESNEXT)
+    ecma_value_t error = jcontext_take_exception ();
+    ecma_property_t *prop_p = ecma_find_named_property (ecma_get_object_from_value (error),
+                                                        ecma_get_magic_string (LIT_MAGIC_STRING_MESSAGE));
+    ecma_free_value (error);
+    const char default_msg[] = "Invalid regular expression";
+    lit_utf8_byte_t *buffer_p = (lit_utf8_byte_t *) default_msg;
+    lit_utf8_size_t size = sizeof (buffer_p) - 1;
+
+    if (prop_p != NULL)
+    {
+      ecma_string_t *message_p = ecma_get_string_from_value (ECMA_PROPERTY_VALUE_PTR (prop_p)->value);
+      JERRY_ASSERT (!ECMA_IS_DIRECT_STRING (message_p));
+      JERRY_ASSERT (ECMA_STRING_GET_CONTAINER (message_p) == ECMA_STRING_CONTAINER_HEAP_ASCII_STRING);
+      buffer_p = ECMA_ASCII_STRING_GET_BUFFER (message_p);
+      size = ((ecma_ascii_string_t *) message_p)->size;
+    }
+
+    literal_p->prop.length = (prop_length_t) size;
+    literal_type = LEXER_STRING_LITERAL;
+    uint8_t *lit_char_buffer_p = (uint8_t *) jmem_heap_alloc_block (size);
+    memcpy (lit_char_buffer_p, buffer_p, size);
+    re_bytecode_p = (re_compiled_code_t *) lit_char_buffer_p;
+#else /* !ENABLED (JERRY_ESNEXT) */
     parser_raise_error (context_p, PARSER_ERR_INVALID_REGEXP);
+#endif /* ENABLED (JERRY_ESNEXT) */
   }
 
-  literal_p->type = LEXER_REGEXP_LITERAL;
+  literal_p->type = literal_type;
```

Comment:
Am I see correctly it will not search the same string literal in case of syntax error?